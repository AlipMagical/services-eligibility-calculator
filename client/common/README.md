# Common

The files contained within this folder are symlinked from `client/common` to `client/service-calculator/src/common` and `client/service-manager/src/common` to share components between the projects.
Ensure that `core.symlinks` is `true` when cloning this repository.
