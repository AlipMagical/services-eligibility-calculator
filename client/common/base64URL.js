/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const base64URLEncode = plainText =>
  btoa(plainText)
    .replace(/=/g, '')
    .replace(/\+/g, '-')
    .replace(/\//g, '_');

export const base64URLDecode = encodedText =>
  atob(encodedText.replace(/-/g, '+').replace(/_/g, '/'));

export default { base64URLEncode, base64URLDecode };
