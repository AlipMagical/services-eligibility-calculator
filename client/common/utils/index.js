/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import ConfirmationPrompt from './ConfirmationPrompt';
import ExternalLink from './ExternalLink';
import FocusOnFirstError from './FocusOnFirstError';
import MapsLink from './MapsLink';
import archivedByDisplayName from './archivedByDisplayName';
import canManageService from './canManageService';
import capitalize from './capitalize';
import getAllOwners from './getAllOwners';
import hasRole from './hasRole';
import lastModifiedByDisplayName from './lastModifiedByDisplayName';
import sendEmail from './sendEmail';
import sharedByDisplayName from './sharedByDisplayName';

export {
  archivedByDisplayName,
  canManageService,
  capitalize,
  ConfirmationPrompt,
  ExternalLink,
  FocusOnFirstError,
  getAllOwners,
  hasRole,
  lastModifiedByDisplayName,
  MapsLink,
  sendEmail,
  sharedByDisplayName
};
