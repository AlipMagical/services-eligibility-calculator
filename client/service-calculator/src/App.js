/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import React, { createContext, useState } from 'react';

import { ANALYTICS_ID } from './common/config';
import { Body } from './components/Body';
import { DataContext } from './DataContext';
import Footer from './components/Footer';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import { Router } from 'react-router-dom';
import ScrollToTop from './common/utils/ScrollToTop';
import { createBrowserHistory } from 'history';

export const FilterContext = createContext();

export const createInitialFilterState = () => ({
  income: undefined,
  interests: {},
  people: {}
});

const history = createBrowserHistory();

export default function App(props) {
  const [analyticsInitialized, setAnalyticsInitialized] = useState(false);
  const [lastSearch, setLastSearch] = useState('');
  const [filterState, setFilterState] = useState(createInitialFilterState());
  const resetFilterState = () => setFilterState(createInitialFilterState());

  if (!analyticsInitialized) {
    ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });
    setAnalyticsInitialized(true);
  }

  return (
    <DataContext.StateProvider>
      <FilterContext.Provider
        value={{
          filterState,
          setFilterState,
          resetFilterState,
          lastSearch,
          setLastSearch
        }}
      >
        <Router history={history}>
          <ScrollToTop />
          <Body />
          <Footer />
        </Router>
      </FilterContext.Provider>
    </DataContext.StateProvider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
