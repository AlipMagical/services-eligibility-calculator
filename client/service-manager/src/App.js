/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import './i18n';

import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import React, { createContext, useState } from 'react';
import {
  getLoggedInUser,
  getMillisUntilExpiration,
  isLoggedIn
} from './common/auth';

import { ANALYTICS_ID } from './common/config';
import { BASEPATH } from './common/config';
import Header from './components/Header';
import PropTypes from 'prop-types';
import ReactGA from 'react-ga';
import Toast from './components/Toast';
import departmentRoutes from './routes/department';
import getUserConfirmation from './getUserConfirmation';
import interestRoutes from './routes/interest';
import serviceRoutes from './routes/service';
import settingsRoutes from './routes/settings';
import { useSettings } from './hooks';

export const AuthContext = createContext();
export const ToastContext = createContext();
export const SettingsContext = createContext();

export default function App(props) {
  ReactGA.initialize(ANALYTICS_ID, { testMode: props.testMode });

  const [loginState, setLoginState] = useState({
    isLoggedIn: isLoggedIn(),
    loggedInUser: getLoggedInUser()
  });

  const onLoginStatusChanged = () => {
    //setTimeout to update when token might expire.
    const millis = getMillisUntilExpiration();
    millis && setTimeout(onLoginStatusChanged, millis);

    setLoginState({
      isLoggedIn: isLoggedIn(),
      loggedInUser: getLoggedInUser()
    });
  };

  const { settings, setSettings } = useSettings();

  const [toastMessage, setToastMessage] = useState(null);

  return (
    <AuthContext.Provider value={{ ...loginState, onLoginStatusChanged }}>
      <SettingsContext.Provider value={{ settings, setSettings }}>
        <ToastContext.Provider value={{ toastMessage, setToastMessage }}>
          <BrowserRouter getUserConfirmation={getUserConfirmation}>
            <Header />
            <div className="container bd-content">
              <Switch>
                {serviceRoutes}
                {settingsRoutes}
                {interestRoutes}
                {departmentRoutes}
                <Redirect from={BASEPATH} exact to={BASEPATH + 'services/'} />
                <Route render={() => <h1>Not Found</h1>} />
              </Switch>
            </div>
            {toastMessage && <Toast message={toastMessage} />}
          </BrowserRouter>
        </ToastContext.Provider>
      </SettingsContext.Provider>
    </AuthContext.Provider>
  );
}

App.propTypes = {
  testMode: PropTypes.bool
};
