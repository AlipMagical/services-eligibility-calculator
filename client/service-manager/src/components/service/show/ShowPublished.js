/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useCallback } from 'react';
import {
  SERVICE_TYPE_ARCHIVED,
  SERVICE_TYPE_PUBLISHED
} from '../../../common/utils/serviceTypes';
import { base64URLDecode, base64URLEncode } from '../../../common/base64URL';

import { BASEPATH } from '../../../common/config';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';
import Show from './Show';
import { canManageService } from '../../../common/utils';
import { defaultLocale } from '../../../Locales';
import { useServiceTranslations } from '../../../hooks';

/**
 * Displays information for a published service.
 */
export default function ShowPublished({ location, history, match }) {
  const serviceUrl = base64URLDecode(match.params.id);
  const handle404 = useCallback(() => history.replace(BASEPATH + 'services/'), [
    history
  ]);
  const translations = useServiceTranslations(serviceUrl, handle404);
  const service = translations[defaultLocale];
  const serviceId = serviceUrl.substring(serviceUrl.lastIndexOf('/') + 1);
  const editLink = (service &&
    (service.draft
      ? {
          pathname: `${BASEPATH}servicedrafts/edit/${base64URLEncode(
            service.draft['@id']
          )}`,
          state: { fromPublished: true }
        }
      : {
          pathname: `${BASEPATH}servicedrafts/create/${base64URLEncode(
            serviceUrl
          )}`,
          state: { serviceHistory: service.history }
        })) || { pathname: '' };

  return (
    <>
      {service && (
        <Helmet>
          <title>{service.name}</title>
        </Helmet>
      )}
      {service && (
        <Show
          {...{
            entityType:
              service && service.archivedDateTime
                ? SERVICE_TYPE_ARCHIVED
                : SERVICE_TYPE_PUBLISHED,
            fromSettings:
              location && location.state && location.state.fromSettings,
            translations,
            entityUrl: serviceUrl,
            serviceId,
            editLink,
            canManageService,
            history,
            serviceHistory: service && service.history
          }}
        />
      )}
    </>
  );
}

ShowPublished.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
};
