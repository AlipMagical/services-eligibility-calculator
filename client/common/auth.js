/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { fetch } from './dataAccess';
import jwt_decode from 'jwt-decode';

export function login(username, password) {
  return fetch('login', {
    method: 'POST',
    body: JSON.stringify({
      username: username,
      password: password
    })
  }).then(handleTokenResponse);
}

export function refreshToken() {
  const refresh_token = localStorage.getItem('refresh_token');
  if (
    refresh_token !== null &&
    jwtWillExpireWithinSeconds(300 /* five minutes */)
  ) {
    return fetch('token/refresh', {
      method: 'POST',
      body: JSON.stringify({
        refresh_token
      })
    }).then(handleTokenResponse);
  }
}

function handleTokenResponse(response) {
  return response.json().then(json => {
    if (!json.token) {
      return false;
    }
    localStorage.setItem('jwt_token', json.token);
    if (json.refresh_token) {
      localStorage.setItem('refresh_token', json.refresh_token);
    } else {
      localStorage.removeItem('refresh_token');
    }
    return true;
  });
}

function jwtWillExpireWithinSeconds(seconds) {
  let encodedToken = localStorage.getItem('jwt_token');
  if (encodedToken != null) {
    let token = jwt_decode(encodedToken);
    return token.exp <= Math.floor(Date.now() / 1000) + seconds;
  }
  return false;
}

export function logout() {
  localStorage.removeItem('jwt_token');
  localStorage.removeItem('refresh_token');
}

export function isLoggedIn() {
  return getEncodedJwtToken() != null;
}

export function getEncodedJwtToken() {
  let encodedToken = localStorage.getItem('jwt_token');
  if (encodedToken != null) {
    let token = jwt_decode(encodedToken);
    if (token.exp <= Math.floor(Date.now() / 1000)) {
      localStorage.removeItem('jwt_token');
      encodedToken = null;
    }
  }
  return encodedToken;
}

export function getMillisUntilExpiration() {
  let encodedToken = getEncodedJwtToken();
  if (encodedToken != null) {
    let expiration = jwt_decode(encodedToken).exp * 1000;
    return expiration - Date.now();
  } else {
    return null;
  }
}

export function getLoggedInUser() {
  let encodedToken = getEncodedJwtToken();
  if (encodedToken != null) {
    return jwt_decode(encodedToken);
  } else {
    return {
      roles: [],
      username: '',
      displayName: '',
      mail: '',
      thumbnailPhoto: ''
    };
  }
}
