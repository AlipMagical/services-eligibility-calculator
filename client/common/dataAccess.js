/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ENTRYPOINT } from './config/entrypoint';
import get from 'lodash/get';
import { getEncodedJwtToken } from './auth';
import has from 'lodash/has';
import mapValues from 'lodash/mapValues';

const MIME_TYPE = 'application/ld+json';

export function fetch(url, options = {}) {
  if ('undefined' === typeof options.headers) options.headers = new Headers();
  if (null === options.headers.get('Accept'))
    options.headers.set('Accept', MIME_TYPE);

  if (
    'undefined' !== options.body &&
    !('undefined' !== typeof FormData && options.body instanceof FormData) &&
    null === options.headers.get('Content-Type')
  ) {
    options.headers.set('Content-Type', MIME_TYPE);
  }

  var jwtToken = getEncodedJwtToken();
  if (jwtToken) {
    options.headers.set('Authorization', 'Bearer ' + jwtToken);
  }

  var fullUrl = new URL(url, ENTRYPOINT);
  return global.fetch(fullUrl.href, options).then(response => {
    if (response.ok) return response;

    if (response.status === 401) {
      throw new Error(response.statusText || 'Authorization Error.');
    }

    return response.json().then(
      json => {
        const error =
          json['hydra:description'] ||
          json['hydra:title'] ||
          'An error occurred.';
        if (!json.violations) throw Error(error);

        let errors = { _error: error };
        json.violations.forEach(violation =>
          errors[violation.propertyPath]
            ? (errors[violation.propertyPath] +=
                '\n' + errors[violation.propertyPath])
            : (errors[violation.propertyPath] = violation.message)
        );

        throw new Error(errors);
      },
      () => {
        throw new Error(response.statusText || 'An error occurred.');
      }
    );
  });
}

export function mercureSubscribe(url, topics) {
  topics.forEach(topic =>
    url.searchParams.append('topic', new URL(topic, ENTRYPOINT))
  );

  return new EventSource(url.toString());
}

export function normalize(data) {
  if (has(data, 'hydra:member')) {
    // Normalize items in collections
    data['hydra:member'] = data['hydra:member'].map(item => normalize(item));

    return data;
  }

  // Flatten nested documents
  return mapValues(data, value =>
    Array.isArray(value)
      ? value.map(v => get(v, '@id', v))
      : get(value, '@id', value)
  );
}

export function extractHubURL(response) {
  const linkHeader = response.headers.get('Link');
  if (!linkHeader) return null;

  const matches = linkHeader.match(
    /<([^>]+)>;\s+rel=(?:mercure|"[^"]*mercure[^"]*")/
  );

  return matches && matches[1] ? new URL(matches[1], ENTRYPOINT) : null;
}
