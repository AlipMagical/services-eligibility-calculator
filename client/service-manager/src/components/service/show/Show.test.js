/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { AuthContext, SettingsContext, ToastContext } from '../../../App';
import { MemoryRouter, Route } from 'react-router-dom';
import { axe, toHaveNoViolations } from 'jest-axe';
import {
  mockService,
  mockServiceAnalyticsJson,
  mockServiceJson
} from '../../../../../testing/mockData';

import React from 'react';
import ShowPublished from './ShowPublished';
import { act } from 'react-dom/test-utils';
import { base64URLEncode } from '../../../common/base64URL';
import { getSettings } from '../../../../../testing/testHelpers';
import { render } from 'react-dom';

expect.extend(toHaveNoViolations);

it('renders show a service view', async () => {
  const container = document.createElement('div');

  fetch.resetMocks();
  fetch
    .doMockOnceIf(
      url =>
        url.includes(
          '/api/services/11ea1d4d-6cd4-b95e-8675-a08cfde838ce?locale=en'
        ),
      mockServiceJson
    )
    .doMockOnceIf(
      url =>
        url.includes(
          '/api/services/11ea1d4d-6cd4-b95e-8675-a08cfde838ce?locale=es'
        ),
      mockServiceJson
    )
    .doMockOnceIf(
      url =>
        url.includes(
          '/api/services/11ea1d4d-6cd4-b95e-8675-a08cfde838ce/analytics'
        ),
      mockServiceAnalyticsJson
    );

  await act(async () => {
    const settings = getSettings();
    return render(
      <MemoryRouter>
        <SettingsContext.Provider value={{ settings }}>
          <AuthContext.Provider
            value={{
              isLoggedIn: true,
              loggedInUser: { username: mockService.lastModifiedBy }
            }}
          >
            <ToastContext.Provider
              value={{ toastMessage: null, setToastMessage: () => {} }}
            >
              <Route
                component={props => (
                  <ShowPublished
                    {...props}
                    match={{
                      params: {
                        id: base64URLEncode(
                          'api/services/11ea1d4d-6cd4-b95e-8675-a08cfde838ce'
                        )
                      }
                    }}
                  />
                )}
              />
            </ToastContext.Provider>
          </AuthContext.Provider>
        </SettingsContext.Provider>
      </MemoryRouter>,
      container
    );
  });

  expect(container).toMatchSnapshot();
  expect(await axe(container)).toHaveNoViolations();
});
