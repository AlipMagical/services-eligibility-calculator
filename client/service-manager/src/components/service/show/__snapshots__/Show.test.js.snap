// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders show a service view 1`] = `
<div>
  <nav
    class="service-header navbar navbar-expand navbar-light sticky-top"
  >
    <div
      class="w-100"
    >
      <h1
        class="float-left"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root material-icons arrow-back"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            arrow_back
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        Accusantium aperiam consequatur.
      </h1>
      <div
        class="float-right"
      >
        <button
          class="btn btn-blue-text"
        >
          Archive
        </button>
        <button
          class="btn btn-blue-text"
        >
          Share
        </button>
        <a
          class="btn btn-primary"
          href="/servicedrafts/create/YXBpL3NlcnZpY2VzLzExZWExZDRkLTZjZDQtYjk1ZS04Njc1LWEwOGNmZGU4MzhjZQ"
          role="button"
        >
          Edit
        </a>
      </div>
    </div>
  </nav>
  <div
    class="language-buttons"
  >
    <button
      class="btn btn-blue-text btn-language-selected"
      type="button"
    >
      English
    </button>
    <button
      class="btn btn-language"
      type="button"
    >
      Spanish
    </button>
  </div>
  <div
    class="row"
  >
    <div
      class="px-0 px-md-3 col-md-8 col-12"
    >
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-body"
        >
          <div
            class="service-title"
          >
            Accusantium aperiam consequatur.
          </div>
          <div
            style="font-size: 16px;"
          >
            <div>
              Rem reprehenderit delectus architecto vero numquam aut repellendus.
            </div>
          </div>
          <div
            class="accepting-applications-future inline box"
          >
            <div>
              <i
                aria-hidden="true"
                class="material-icons"
              >
                schedule
              </i>
            </div>
            <div>
              Accepting applications starting 
              Apr 27, 2020
            </div>
          </div>
        </div>
      </div>
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          What is the service?
          <button
            aria-label="minimize"
            class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              remove
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div>
                Impedit non assumenda similique. Voluptates sunt tempora voluptas accusantium alias reprehenderit veritatis. Eos doloribus et magnam est enim totam necessitatibus.
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              Department
              :
            </div>
            <div
              class="text-break col"
            >
              Housing and Human Services
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              Application window
              :
            </div>
            <div
              class="text-break col"
            >
              Apr 27, 2020
               - 
              Sep 3, 2020
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              Relates to
              :
            </div>
            <div
              class="text-break col"
            >
              <span>
                Wellness
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          Who is eligible?
          <button
            aria-label="minimize"
            class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              remove
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="card-body"
        />
      </div>
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          How do I apply?
          <button
            aria-label="minimize"
            class="MuiButtonBase-root MuiIconButton-root card-collapse-icon float-right material-icons align-middle"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              remove
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              Learn more
              :
            </div>
            <div
              class="col"
            >
              <a
                href="https://marquardt.biz/possimus-cupiditate-iure-quo-sit-ratione.html"
                rel="noopener noreferrer"
                target="_blank"
              >
                https://marquardt.biz/possimus-cupiditate-iure-quo-sit-ratione.html
              </a>
            </div>
          </div>
          <hr />
          <div
            class="instructions"
          >
            First, you'll need
            :
          </div>
          <div
            class="row"
          >
            <div
              class="text-break col"
            >
              <div>
                Totam porro consequatur molestias recusandae non hic. Et libero aperiam commodi dolor.
              </div>
            </div>
          </div>
          <hr />
          <div
            class="instructions"
          >
            When you're ready, apply
            :
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              <i
                aria-hidden="true"
                class="material-icons apply-icon align-middle"
              >
                phone
              </i>
              <span
                class="align-middle"
              >
                Call:
              </span>
            </div>
            <div
              class="col"
            >
              <a
                href="tel:827.937.7299"
                rel="noopener noreferrer"
                target="_blank"
              >
                827.937.7299
              </a>
               
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              <i
                aria-hidden="true"
                class="material-icons apply-icon align-middle"
              >
                email
              </i>
              <span
                class="align-middle"
              >
                Email:
              </span>
            </div>
            <div
              class="col"
            >
              <a
                href="mailto:ebert.harley@prosacco.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                ebert.harley@prosacco.com
              </a>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              <i
                aria-hidden="true"
                class="material-icons apply-icon align-middle"
              >
                computer
              </i>
              <span
                class="align-middle"
              >
                Online:
              </span>
            </div>
            <div
              class="col"
            >
              <a
                href="http://www.oconner.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                http://www.oconner.com/
              </a>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="label col-md-4"
            >
              <i
                aria-hidden="true"
                class="material-icons apply-icon align-middle"
              >
                directions_walk
              </i>
              <span
                class="align-middle"
              >
                In person:
              </span>
            </div>
            <div
              class="col"
            >
              <a
                href="https://www.google.com/maps/search/?api=1&query=4070%20Torphy%20Isle%2C%20Stokesburgh%2C%20VT%2032336"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div>
                  4070 Torphy Isle, Stokesburgh, VT 32336
                </div>
              </a>
              <br />
              <div>
                A eaque tempora quia ex id qui assumenda.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-0 px-md-3 col-md-4 col-12"
    >
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          Owners
        </div>
        <div
          class="card-body"
        >
          <p>
            <i>
              Owners can edit and archive this service.
            </i>
          </p>
          <div>
            brittany96@domain.com
          </div>
        </div>
      </div>
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          Analytics
          <a
            class="analytics-more"
            href="https://analytics.google.com/analytics/web/#/report/content-pages/a154731980w218295360p208192424/_u.dateOption=last30days&_u.hasComparison=true&_r.drilldown=analytics.pagePath:~2Fservices~2FL2FwaS9zZXJ2aWNlcy8xMWVhMWQ0ZC02Y2Q0LWI5NWUtODY3NS1hMDhjZmRlODM4Y2U&explorer-table.plotKeys=%5B%5D&explorer-table.secSegmentId=analytics.pageTitle/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="btn btn-blue-text"
            >
              More 
              <i
                class="material-icons"
              >
                open_in_new
              </i>
            </span>
          </a>
        </div>
        <div
          class="card-body"
        >
          <div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <span
                  class="analytics-pageviews"
                >
                  42
                </span>
                <span>
                  views in last 30 days
                </span>
              </div>
              <div
                class="analytics-trending analytics-trending-up"
              >
                <i
                  class="material-icons"
                >
                  trending_up
                </i>
                100
                %
              </div>
            </div>
            <div
              class="row"
            />
            <div
              class="row"
            >
              <div
                class="col"
              >
                <span
                  class="analytics-clicks"
                >
                  10
                </span>
                <span>
                  clicked “Learn More”
                </span>
              </div>
              <div
                class="analytics-trending"
              >
                <i
                  class="material-icons"
                >
                  trending_down
                </i>
                30
                %
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <span
                  class="analytics-clicks"
                >
                  23
                </span>
                <span>
                  clicked “Share”
                </span>
              </div>
              <div
                class="analytics-trending"
              >
                <i
                  class="material-icons"
                >
                  trending_flat
                </i>
                0
                %
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="service-card card"
        tabindex="-1"
      >
        <div
          class="card-header"
        >
          History
        </div>
        <div
          class="card-body"
        >
          <div
            class="history-card"
          >
            No history available yet
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
